### PR TITLE
fix: local declaration mutations are promoted from statement to block level mutations

### PR DESCRIPTION
**Pull request by dupdob**
Fix issues #1423 (and related). Statement level mutations are promoted as blocklevel mutations for LocalDeclaration statement, to prevent them being hidden in a subscope when mutated.

### CODE REVIEW

**Comment by Mobrockers** [comment]
Quick fix as always 💖

Btw I changed the target branch to master. We have decided to move the main branch to develop to focus on a 1.0 release and to make sure no new features make it on to master, which has the side effect that the dev branch is automatically selected for pull requests. Bug fixes will still go to master so we can still release fixes without new features.

**Comment by dupdob** [comment]
I failed to understand your remark about branches at first! Now I do. I did not notice that Github selected `v1.0' as the target.
I was not even aware there was v 1 branch.

I'll try to find a fix for the #1426, I can not reproduce for now. I can give details on the slack if interested.

File: src/Stryker.Core/Stryker.Core.UnitTest/Mutants/MutantOrchestratorTests.cs [hunk 1]
```diff
@@ -886,6 +886,43 @@ public void ShouldMutatePropertiesAsArrowExpression()
             ShouldMutateSourceToExpected(source, expected);
         }
 
+
+        [Fact]
+        public void ShouldNotExhibitIssue1423()
```
Comment (Stijn-Rutten) [inline src/Stryker.Core/Stryker.Core.UnitTest/Mutants/MutantOrchestratorTests.cs:891]: Could you maybe add a more descriptive name to this test? I would like a clearer name that allows me to see the problem at first glace. Maybe reference the issue in the description so when we need more info we can reference that.

File: src/Stryker.Core/Stryker.Core/Mutants/NodeOrchestrators/LocalDeclarationOrchestrator.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            if (node.IsConst)
+            {
+                // don't mutate const declaration statement
+                return node;
+            }
+
+            var result = base.OrchestrateChildrenMutation(node, context);
+            // statement level mutations need to be changed to block level
+            context.BlockLevelControlledMutations.AddRange(context.StatementLevelControlledMutations);
+            context.StatementLevelControlledMutations.Clear();
```
Comment (richardwerkman) [inline src/Stryker.Core/Stryker.Core/Mutants/NodeOrchestrators/LocalDeclarationOrchestrator.cs:24]: I'm thinking this is the solution right? Can you explain why this solves our issue?

**Pull request merged**